### ISSUE TITLE
`brownie test --coverage` produces empty json file - Tested with every brownie release since 1.15.2 - With A / B Test

### ISSUE THREAD

**Issue by GalloDaSballo** [eth-brownie/brownie#1562]
The following issue shows 2 sets of code that are really similar, however one version of the code works with `--coverage` as intended, while the second breaks the system for some reason

### Environment information

* `brownie` Version: 1.15 upwards to 1.19.0
* `ganache-cli` 6.12.2
* `solc` Version: 0.8.12
* Python Version: 3.9
* OS: osx/linux/win: Mac

### What was wrong?

Running tests with `--coverage` flag produces empty json object.

I've tried with every brownie version since 1.15.2 and all tests run and pass properly, but coverage `.json` file is a bunch of empty json data.

Please include information like:

Has been tried by multiple devs, with the code in this repository:
https://github.com/GalloDaSballo/badger-onchain-rewards/tree/feat-dust-handling

Coverage does work for:
https://github.com/GalloDaSballo/badger-onchain-rewards

* what command you ran -> `brownie test --coverage`
* the code that caused the failure (see [this link](https://help.github.com/articles/basic-writing-and-formatting-syntax/) for help with formatting code)
* full output of the error you received

### How can it be fixed?

I have no clue but assume that the differences between the two code-bases (which are very minor) should help you shed light into this pretty obscure behaviour

TL;DR
main -> Coverage works
feat-dust-handling -> Coverage doesnt' work and sets up an empty file that looks like this:
```
{
  "coverage": {},
  "highlights": {
    "branches": {},
    "statements": {}
  },
  "sha1": {}
}
```
